modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -172,7 +173,7 @@ public static <RowT> ExecutionServiceImpl<RowT> create(
                 dependencyResolver,
                 (ctx, deps) -> new LogicalRelImplementor<>(
                         ctx,
-                        new HashFunctionFactoryImpl<>(sqlSchemaManager, handler),
+                        new HashFunctionFactoryImpl<>(ctx.getRootSchema().unwrap(IgniteSchema.class), handler),
```

Comment:
Not quiet, seems, you can reuse function for multiple tables.
 `public RowHashFunction<T> create(int[] fields, int tableId) {`

Reply:
I didn't catch what is your suggestion.
A query may use multiple tables, e.g. in joins. At a moment Factory is created, there is no tableID.

Reply:
I've changed this, because HashFunctionFactoryImpl don't aware of schema versions, but sqlSchemaManager methods require schema version for getting table descriptor. 
Moreover, schema version is fixed for the query and SchemaManager is not needed anymore.